flake.nix

```diff
@@ -0,0 +1,40 @@
+{
+    description = ''
+        A near drop-in replacement for rm that uses the FreeDesktop trash bin.
+        Written in the D programming language using only D's Phobos standard library, and can be compiled with any recent D compiler. This includes GCC, so `trash-d` should run on any *NIX platform that GCC supports.
+    '';
+
+    inputs = {
+        nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+        flake-utils.url = "github:numtide/flake-utils";
+    };
+
+    outputs = { self, nixpkgs, flake-utils }: flake-utils.lib.eachDefaultSystem (system:
+        let pkgs = nixpkgs.legacyPackages.${system}; in
+        {
+            defaultPackage = pkgs.stdenv.mkDerivation {
+                name = "trash-d";
+                src = self;
+
+                propagatedBuildInputs = with pkgs; [
+                    (dmd.overrideAttrs (old : {
+                        doCheck = false;
+                    }))
+                    gcc
+                    rake
+                    dub
+                    ronn
+                ];
+
+                buildPhase = "rake";
+
+                installPhase = ''
+                    mkdir -p $out/bin
+                    install -t $out/bin build/trash
```

Comment:
You should set the permissions as well when using `install` with the `-m` flag, and you can skip the `mkdir -p` by using `install -D`.

Reply:
The examples I looked at didn't use any permissions, and when I tried changing the permissions in `flake.nix` it didn't reflect on the produced executable. I'm guessing Nix does some stuff behind the scenes here because the file also seems to be owned by root.
```
Permissions Size User Date Modified Name
.r-xr-xr-x  3,7M root  1 jan  1970  trash
```
Let me know if you know more about this, I can't really find an answer to this so fast